### ISSUE TITLE
New Popunder filters

### ISSUE THREAD

**Issue by IDKwhattoputhere** [gorhill/uBlock#1094]
Popunders on PornHub, YouPorn and other PHN sites are not being blocked in Chrome (Firefox is fine). It opens the video in a new tab which gets closed by uBO and now only the popunder is open and loads. Can this be blocked by the new popunder feature in uBO? Or is inline script tag filtering needed?

PS: I think this only is because they switched the domain.
Old: ebocornac.com
New: slopeaota.com

**Comment by gorhill** [comment]
Can you details the exact step to reproduce? (with URLs etc.)

**Comment by IDKwhattoputhere** [comment]
Sure.
URLs (e.g.):
http://www.youporn.com/
http://www.pornhub.com/

Click on a video thumbnail. A new tab with the video page opens but gets closed by uBO (must be a EasyList filter). The original tab now loads the popunder.

Short demonstration video (loaded really slow so you can see the redirect happening): https://u.pomf.io/elsoxc.mp4

**Comment by gorhill** [comment]
I could not reproduce popunders. There were popups (if clearing cookies), which were properly blocked with EayList's:

    http://*?*=$domain=extremetube.com|pornhub.com|redtube.com|spankwire.com|tube8.com|youporn.com|youporngay.com,third-party,popup

Since you can reproduce, can you see if this work:

    ||slopeaota.com^$document,popunder

**Comment by IDKwhattoputhere** [comment]
I set up a new profile with the latest Chromium version and uBO in default configuration and the popunder still occurred. I added your filter and it blocked it just fine. Only weird thing is it doesn't seem to work in my normal Chrome profile. The filter closes the video page and the popunder and since those where the only tabs quits the browser. I tested it with both uBO instances having the same settings but no difference.
Regular Chrome profile: https://u.pomf.io/paqgmm.mp4
Fresh Chromium profile: https://u.pomf.io/fmrxlb.mp4
I can't tell how this is possible since I don't have any other blocker extenions enabled. I hope it only applies to me and others will be fine.

**Comment by IDKwhattoputhere** [comment]
I set up a new profile with the latest Chromium version and uBO in default configuration and the popunder still occurred. I added your filter and it blocked it just fine. Only weird thing is it doesn't seem to work in my normal Chrome profile. The filter closes the video page and the popunder and since those where the only tabs quits the browser. I tested it with both uBO instances having the same settings but no difference.
Regular Chrome profile: https://u.pomf.io/paqgmm.mp4
Fresh Chromium profile: https://u.pomf.io/fmrxlb.mp4
I can't tell how this is possible. I hope it only applies to me and others will be fine.

**Comment by gorhill** [comment]
> It closes the video page and the popunder and since those where the only tabs quits the browser

Use the logger with the _All_ entry selected in the drop down. See what it reports when this occurs.

**Comment by IDKwhattoputhere** [comment]
I can't see anything of interest. It only shows that it blocked the popunder.
![ph](https://cloud.githubusercontent.com/assets/16030872/11962088/60e6db7e-a8dd-11e5-818c-6e4d1c170bd0.png)
![yp](https://cloud.githubusercontent.com/assets/16030872/11962089/60e9c082-a8dd-11e5-885b-f7315986d9c1.png)

**Comment by gorhill** [comment]
Ok, start from scratch, and explain step by step what I need to do to reproduce a popunder, including URLs and all. **Important:** if it is not a popunder, do **not** report here, this is for EasyList manitainers to take care.

**Comment by IDKwhattoputhere** [comment]
Go to http://www.pornhub.com/ and click on e.g. Tube8 at the top. Real site gets blocked as a popup and popunder loads. ||slopeaota.com^$document,popunder is in my filters.
It's basically all there in the video. For some reason uBO doesn't count it as a popunder but it is.

![2015](https://cloud.githubusercontent.com/assets/16030872/11965069/75395a88-a8f2-11e5-8ac8-fd7bd97af7ed.png)

**Comment by gorhill** [comment]
I just can't reproduce a popunder, others will have to come forward with their findings.

From the log, I see that a legitimate target got closed. First, it does not happen on my side if I merely click, only if I middle-click. Second, the fact that the legitimate target got closed, is because of a very broad EasyList filter, this also happens with ABP if middle-clicking. Surely a narrower filter would work better. Third, as a consequence of the target page being closed as a popup (false positive), no popunder will be processed: uBO will never look for popunder if a popup was blocked, as it is assumed you can't possibly have both a popup and a popunder at the same time from the same source window.

So in short, the false positive on the popup prevents the popunder filter from being hit. The false positive is caused by EasyList overly broad popup filter, seen in your logger.

**Comment by gorhill** [comment]
You could try to create exception filters for what might be legitimate targets from each web site. For example, using the current case:

    @@||tube8.com^$popup,domain=extremetube.com|pornhub.com|redtube.com|spankwire.com|youporn.com|youporngay.com

One would have to create as many filters as there are legitimate targets on these sites:
    @@||extremetube.com^$popup,domain=pornhub.com|redtube.com|spankwire.com|tube8.com|youporn.com|youporngay.com

Etc.

See if this works.

**Comment by IDKwhattoputhere** [comment]
Yes, if I whitelist e.g. Tube8 popups it works when I click Tube8 and so on.
I reported the popups as incorrectly blocked content on the forums too since they are all legitimate.

**Comment by IDKwhattoputhere** [comment]
I tried 1.5.0 in Chromium and no popunders where had (yay!) except on tube8.com. You put tube8.* in the filters. Why *? And are you sure wildcards work in there? Since uBO set InfNumFastPopsExpire on all sites to 2 minutes in the future but not on tube8.com (as far as I could see). And is 2 minutes enough or could it happen that the site reloads something and uBO doesn't apply the script again? Sorry for potentially dumb questions.

**Comment by gorhill** [comment]
> And are you sure wildcards work in there?

Yes, except that there was a typo in the code, preventing these entity-based filters from working. Fixed in a7901cd97e5faabfc10bf8b151f5645297840582. Using `tube8.*` avoid the need to declare `tube8.com,tube8.es,tube8.fr` etc.

**Comment by gorhill** [comment]
Oops, in the commit message of a7901cd97e5faabfc10bf8b151f5645297840582 I used a link to your GitHub account instead of a link to your comment above (bad right-click copy-link).

**Comment by IDKwhattoputhere** [comment]
> [...]I used a link to your GitHub account[...]

No problem.
> Using tube8.* avoid the need to declare tube8.com,tube8.es,tube8.fr etc.

I didn't know they had other TLDs.
Updated the lists and now it also works on Tube8. Very nice. :)

**Comment by flyingzebras** [comment]
I can confirm that slopeaota.com now also activates popunders in Firefox.

**Comment by IDKwhattoputhere** [comment]
@flyingzebras Even with uBO installed, right? This happens to me in Firefox on pornhub.com but not when I create a fresh profile. I even purged all filter list caches and made an update.

**Comment by flyingzebras** [comment]
@IDKwhattoputhere Yes. I asked Peter Lowe to add the site to his filter.

**Comment by IDKwhattoputhere** [comment]
@flyingzebras Good idea. Thank you. I think the problem has something to do with the way uBO defuses their popunders. I get 2 error messages in the console when using my regular Firefox profile:

    TypeError: localStorage is null
    TypeError: sessionStorage is null
I'm not getting these errors in a fresh profile. I think uBO has to modify the storage. I assume dom.storage.enabled has to be true. I set it to false because it is a minor privacy risk. Could you check that value in about:config?

**Comment by gorhill** [comment]
@flyingzebras `slopeaota.com` is already blocked through the filter:

    |http://$domain=pornhub.com|redtube.com|redtube.com.br|tube8.com|tube8.es|tube8.fr|youporn.com|youporngay.com,third-party,script

@IDKwhattoputhere These errors are not triggered by uBO, uBO checks whether the storages are valid before using them.

**Comment by IDKwhattoputhere** [comment]
@gorhill The page doesn't need to load anything from slopeaota.com. It opens the popunder with inline scripts or whatever it's called.

How come the issue resolves then when I set dom.storage.enabled to true? The error messages disappear and the popunders do too.

**Comment by gorhill** [comment]
See [uBO's code](<https://github.com/gorhill/uBlock/blob/71c2a8fd6b647b63d0aab3de6add95b51b76a5e2/assets/ublock/resources.txt#L407>). The console where the message is printed tells you that the faulty script code comes from <http://cdn1b.static.pornhub.phncdn.com/www-static/js/premium/premium-modals.js?cache=2016011404>.

**Comment by IDKwhattoputhere** [comment]
Yes, I see now. Still, there is no script from slopeaota.com and dom.storage.enabled caused the issue for me.

**Comment by gorhill** [comment]
Given I don't know how to reproduc, does the following work? `||slopeaota.com^$document,popunder,popup`

**Comment by IDKwhattoputhere** [comment]
Yes, works like a charm. The defuser simply doesn't work since the storage is not available I guess. So that filter solves the issue. Thanks.

**Comment by flyingzebras** [comment]
@gorhill Your suggested filter worked for me too.

**Comment by flyingzebras** [comment]
@gorhill @IDKwhattoputhere I just noticed this:


re-directme.com used by pornhub.com et al.

Example: http://re-directme.com/joy2.html?/?partner=p13238p31372p12e2&promo=Doublepimp_pops_SE_Pornhub

This will block it, I think.

```||re-directme.com^$document,popunder,popup```


I reported re-directme.com to ```Peter Lowe’s Ad server list‎ (pgl.yoyo.org)```

**Comment by IDKwhattoputhere** [comment]
Where did you find that? I don't see it on any PHN site.

**Comment by flyingzebras** [comment]
@IDKwhattoputhere pornhub.com use it.

**Comment by IDKwhattoputhere** [comment]
So slopeaota.com redirects to re-directme.com? That would explain why I don't see it.

**Comment by flyingzebras** [comment]
@IDKwhattoputhere ```||slopeaota.com^$document,popunder,popup``` have been added to uBO filters. Just update the 3rd-party filter list.

**Comment by flyingzebras** [comment]
@gorhill trhunt.com is the new popunder used by spankwire.com (Pornhub NETWORK site). trhunt.com then connects to thrnt.com which later connects to traffichunt.com. Can you please add them to the uBlock filter list?

**Comment by IDKwhattoputhere** [comment]
Spankwire is not part of the PHN division that e.g. Youporn is part of. The method they use is not as nasty. It doesn't require uBO specific filter rules.
https://forums.lanik.us/viewtopic.php?f=62&t=27891

**Comment by flyingzebras** [comment]
@IDKwhattoputhere Thank you for pointing this out.

**Comment by uBlock-user** [comment]
@gorhill Lately, pornhub is inserting an additional cookie named `FPSRN` to trigger popups FYI

Deleting this particular cookie does stop the popups from triggering.

**Comment by uBlock-user** [comment]
well adding - `pornhub.com##script:inject(abort-on-property-write.js, UAParser)` takes care of it.
